nncf/common/graph/graph.py

```diff
@@ -294,9 +434,130 @@ def topological_sort(self) -> List[NNCFNode]:
                                                     key=lambda x: self._nx_graph.nodes[x][NNCFGraph.ID_NODE_ATTR])]
 
     def dump_graph(self, path: str):
-        """
-        Writes graph in .dot format.
-
-        :param path: Path to save.
-        """
-        nx.drawing.nx_pydot.write_dot(self._nx_graph, path)
+        nx.drawing.nx_pydot.write_dot(self.get_graph_for_structure_analysis(), path)
+
+    def visualize_graph(self, path: str):
+        out_graph = self._get_graph_for_visualization()
+        nx.drawing.nx_pydot.write_dot(out_graph, path)
+        try:
+            A = to_agraph(out_graph)
+            A.layout('dot')
+            png_path = os.path.splitext(path)[0]+'.png'
+            A.draw(png_path)
+        except ImportError:
+            nncf_logger.warning("Graphviz is not installed - only the .dot model visualization format will be used. "
+                                "Install pygraphviz into your Python environment and graphviz system-wide to enable "
+                                "PNG rendering.")
+
+    def get_graph_for_structure_analysis(self, extended=False) -> nx.DiGraph:
+        """The graph to dump has certain node attributes omitted, compared to the graph stored
```

Comment:
Please, format the docstring according to the Style Guide.

Reply:
Done